# Game Theory Measurement

### MSP 9.18.2017

Checking in on game theory by analyzing the decision to start and sit players over the course of the year in A League Of Their Own (ALOTO).

In [ ]:
from IPython.display import HTML
HTML('''<script>code_show=true; function code_toggle() { if (code_show){ $('div.input').hide();} else { $('div.input').show(); } code_show = !code_show} $( document ).ready(code_toggle);</script><form action="javascript:code_toggle()"><input type="submit" value="Click here to toggle on/off the raw code."></form>''')

In [2]:
# boilerplate imports

import numpy as np
import matplotlib.pyplot as plt
import matplotlib.cm as cm
%matplotlib inline

import matplotlib as mpl



In [13]:
BatMLB = np.genfromtxt('aloto-batting-mlb.dat',\
                  dtype={'names': ("Team","BA","HR","R","RBI","SB"),\
                         'formats': ('S26','f4','f4','f4','f4','f4')},\
                 skip_header=0,delimiter='\t')

BatF = np.genfromtxt('aloto-batting-fantasy.dat',\
                  dtype={'names': ("Team","BA","HR","R","RBI","SB"),\
                         'formats': ('S26','f4','f4','f4','f4','f4')},\
                 skip_header=0,delimiter='\t')

PitchMLB = np.genfromtxt('aloto-pitching-mlb.dat',\
                  dtype={'names': ("Team","ERA","K","S","W","WHIP"),\
                         'formats': ('S26','f4','f4','f4','f4','f4')},\
                 skip_header=0,delimiter='\t')

PitchF = np.genfromtxt('aloto-pitching-fantasy.dat',\
                  dtype={'names': ("Team","ERA","K","S","W","WHIP"),\
                         'formats': ('S26','f4','f4','f4','f4','f4')},\
                 skip_header=0,delimiter='\t')





#### Compile Hitters

This cell only needs to be run in the event that player ID numbers change, or new players are added.

In [14]:
for indx in range(0,len(BatMLB['Team'])):
    print('{0:24s}{1:8.3f}{2:8.3f}{3:8.3f}'.format(BatMLB['Team'][indx],BatMLB['BA'][indx],BatF['BA'][indx],BatF['BA'][indx]-BatMLB['BA'][indx]))

80 Proofers                0.258   0.250  -0.008
Dingers                    0.261   0.257  -0.004
dodger blues               0.271   0.265  -0.006
Flatulent Fellas           0.272   0.265  -0.007
Free True Outcomes         0.275   0.272  -0.003
Magic Johnson              0.282   0.284   0.002
Pack Attack                0.267   0.268   0.001
Ryan's Rebuild             0.281   0.275  -0.006
The Clown Inquisition      0.279   0.272  -0.007
The Wiffleball Legends     0.256   0.253  -0.003
Visalia Mets               0.268   0.271   0.003
Wrath of Khan              0.281   0.281   0.000


In [20]:
metric = 'HR'

for indx in range(0,len(BatMLB['Team'])):
    print('{0:24s}{1:8.3f}{2:8.3f}{3:8.3f}'.format(\
            BatMLB['Team'][indx],BatMLB[metric][indx],BatF[metric][indx],(BatF[metric][indx]-BatMLB[metric][indx])/BatMLB[metric][indx]))

80 Proofers              315.000 284.000  -0.098
Dingers                  311.000 263.000  -0.154
dodger blues             316.000 273.000  -0.136
Flatulent Fellas         276.000 207.000  -0.250
Free True Outcomes       350.000 311.000  -0.111
Magic Johnson            333.000 300.000  -0.099
Pack Attack              264.000 229.000  -0.133
Ryan's Rebuild           314.000 265.000  -0.156
The Clown Inquisition    291.000 270.000  -0.072
The Wiffleball Legends   316.000 186.000  -0.411
Visalia Mets             347.000 250.000  -0.280
Wrath of Khan            238.000 184.000  -0.227


### Pitching Metrics

In [17]:
metric = 'ERA'

for indx in range(0,len(PitchMLB['Team'])):
    print('{0:24s}{1:8.3f}{2:8.3f}{3:8.3f}'.format(\
            PitchMLB['Team'][indx],PitchMLB[metric][indx],PitchF[metric][indx],PitchF[metric][indx]-PitchMLB[metric][indx]))

80 Proofers                3.930   4.260   0.330
Dingers                    3.760   4.250   0.490
dodger blues               4.160   4.410   0.250
Flatulent Fellas           4.120   4.020  -0.100
Free True Outcomes         3.880   3.980   0.100
Magic Johnson              4.170   4.510   0.340
Pack Attack                3.650   4.220   0.570
Ryan's Rebuild             3.640   3.940   0.300
The Clown Inquisition      2.870   2.990   0.120
The Wiffleball Legends     3.910   3.880  -0.030
Visalia Mets               4.470   4.450  -0.020
Wrath of Khan              3.660   3.810   0.150


In [18]:
metric = 'WHIP'

for indx in range(0,len(PitchMLB['Team'])):
    print('{0:24s}{1:8.3f}{2:8.3f}{3:8.3f}'.format(\
            PitchMLB['Team'][indx],PitchMLB[metric][indx],PitchF[metric][indx],PitchF[metric][indx]-PitchMLB[metric][indx]))

80 Proofers                1.280   1.270  -0.010
Dingers                    1.210   1.260   0.050
dodger blues               1.280   1.330   0.050
Flatulent Fellas           1.270   1.270   0.000
Free True Outcomes         1.240   1.230  -0.010
Magic Johnson              1.370   1.380   0.010
Pack Attack                1.230   1.340   0.110
Ryan's Rebuild             1.220   1.290   0.070
The Clown Inquisition      1.060   1.080   0.020
The Wiffleball Legends     1.280   1.280   0.000
Visalia Mets               1.380   1.360  -0.020
Wrath of Khan              1.230   1.270   0.040


### Case Study: Finals Matchups.

Wrath of Khan	0.289	18	66	69	17
Wrath of Khan-M	0.284	24	82	80	20

Ryan's Rebuild	0.269	26	82	75	6
Ryan's Rebuild-M	0.269	34	109	104	7

In this matchup, we basically both started everyone we had on our team at a given time, with some minor cycling. It was a 'classic' BA-SB/HR-R-RBI split, unfortunately for me.



The Clown Inquisition	0.281	20	98	65	11
The Clown Inquisition-M	0.278	30	118	95	18

Free True Outcomes	0.290	26	70	90	9
Free True Outcomes-M	0.269	28	78	104	7

Wrath of Khan	4.06	139	9	13	1.33
Wrath of Khan-M	3.78	133	13	13	1.33

Ryan's Rebuild	3.78	108	2	8	1.08
Ryan's Rebuild-M	3.03	167	1	11	0.99

We both mis-managed our ERA a little bit, but I sadly managed my (too late realized) sky-high WHIP. Saves were yet again mismanaged, the perils of owning a fourth closer for too long, but wins were assess appropriately.


The Clown Inquisition	3.15	138	12	11	1.08
The Clown Inquisition-M	1.68	121	12	9	0.85

Free True Outcomes	4.26	138	2	12	1.28
Free True Outcomes-M	3.55	114	3	10	1.21

This one kills--the entire matchup came down to a single strikeout
